scripts/ci/docker_upgrade.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+set -e
+
+function error() {
+  echo >&2 "ERROR: ${1} Aborting." && false
+}
+
+function warning() {
+  echo >&2 "WARNING: ${1}"
+}
+
+[[ $# == 3 ]] || error "Expected exactly 3 parameters: ${0} <IMAGE_NAME> <IMAGE_VARIANT> <DOCKER_REPOSITORY>"
+
+IMAGE_NAME=${1}
+IMAGE_VARIANT=${2}
+DOCKER_REPOSITORY=${3}
+
+check_version() {
+  echo "-- check_version"
+
+  DOCKERFILE="scripts/docker/${IMAGE_NAME}/Dockerfile.${IMAGE_VARIANT}"
+
+  # continue running the whole workflow only if corresponding Dockerfile was changed.
+  if ! git diff --name-only origin/develop HEAD  -- | grep -e "^${DOCKERFILE}\$"
+  then
+    echo "${DOCKERFILE} was not changed. Nothing to do."
+    exit 0
+  fi
+
+  PREV_VERSION=$(git diff origin/develop HEAD -- "${DOCKERFILE}" | grep -e '^\s*-LABEL\s\+version=".*"\s*$' | awk -F'"' '{ print $2 }')
+  NEXT_VERSION=$(git diff origin/develop HEAD -- "${DOCKERFILE}" | grep -e '^\s*+LABEL\s\+version=".*"\s*$' | awk -F'"' '{ print $2 }')
+
+  [[ $NEXT_VERSION != "" ]] || error "No version label defined in Dockerfile. You may need to add 'LABEL version' in '${DOCKERFILE}'."
+
+  [[ $PREV_VERSION != "" ]] || {
+    warning "no previous version found. Will set \$PREV_VERSION = 0."
+    PREV_VERSION=0
+  }
+
+  if [[ $((PREV_VERSION + 1)) != $((NEXT_VERSION)) ]]; then
+    error "Version label in Dockerfile was not incremented. You may need to change 'LABEL version' in '${DOCKERFILE}'."
+  fi
+}
+
+build_docker() {
+  echo "-- build_docker"
+
+  docker build "scripts/docker/${IMAGE_NAME}" --file "scripts/docker/${IMAGE_NAME}/Dockerfile.${IMAGE_VARIANT}" --tag "${IMAGE_NAME}" || docker build "scripts/docker/${IMAGE_NAME}" --file "scripts/docker/${IMAGE_NAME}/Dockerfile.${IMAGE_VARIANT}" --tag "${IMAGE_NAME}"
```

Comment:
I think it would still work wrapped like this:
```suggestion
  docker build "scripts/docker/${IMAGE_NAME}" --file "scripts/docker/${IMAGE_NAME}/Dockerfile.${IMAGE_VARIANT}" --tag "${IMAGE_NAME}" ||
  docker build "scripts/docker/${IMAGE_NAME}" --file "scripts/docker/${IMAGE_NAME}/Dockerfile.${IMAGE_VARIANT}" --tag "${IMAGE_NAME}"
```